packages/calypso-e2e/src/lib/blocks/file-block.ts

```diff
@@ -0,0 +1,66 @@
+import { Page, ElementHandle, Response, Frame } from 'playwright';
+
+const selectors = {
+	block: '.wp-block-file',
+	fileInput: '.components-form-file-upload input[type="file"]',
+	loadingAnimation: '.components-animate__loading.is-transient',
+};
+
+/**
+ * Represents the File block.
+ */
+export class FileBlock {
+	static blockName = 'File';
+	block: ElementHandle;
+
+	/**
+	 * Constructs an instance of this block.
+	 *
+	 * @param {ElementHandle} block Handle referencing the block as inserted on the Gutenberg editor.
+	 */
+	constructor( block: ElementHandle ) {
+		this.block = block;
+	}
+
+	/**
+	 * Uplaods the target file at the supplied path to WPCOM.
+	 *
+	 * @param {string} path Path to the file on disk.
+	 */
+	async upload( path: string ): Promise< void > {
+		// Obtain the page that this block resides on by first obtaining the frame then the page on which
+		// the frame belongs to.
+		const frame = ( await this.block.ownerFrame() ) as Frame;
+		const page = frame.page() as Page;
```

Comment:
To keep the signature consistent across File, Image and Audio blocks I have decided that obtaining the underlying `page` via this method was preferable.